qiskit/utils/deprecation.py

```diff
@@ -37,19 +37,21 @@ def deprecate_function(msg, stacklevel=2, docstring_warning=True):
     Args:
         msg (str): Warning message to emit.
         stacklevel (int): The warning stackevel to use, defaults to 2.
-        docstring_warning (bool): If `True`, extends the docstring with a warning box.
+        docstring_warning (bool): If `True`, extends the docstring with a warning
+           box. Default: True
 
     Returns:
         Callable: The decorated, deprecated callable.
     """
 
     def decorator(func):
+        if docstring_warning:
+            docstr = func.__doc__.split("")
+            docstr.insert(1, "\n.. warning::\n    msg")
+            func.__doc__ = "\n".join(docstr)
```

Comment:
You need to make sure the indentation levels match everywhere, including the `.. warning` bit.  The first line of a docstring usually has no indentation, but all subsequent lines are indented by some variable amount (depends if it's a class method or a function, for example).  You'll need to make sure you're setting the right indent on everything to match up with the later lines, not the first.

PEP 257 has a simple implementation of the de-indent algorithm used by formatters, so you can use something like that to "reverse" the procedure when adding an indentation, possibly: https://peps.python.org/pep-0257/.

Reply:
aha! That explains a lot. Thanks!